app/metrics.yaml

```diff
@@ -1270,7 +1270,7 @@ metrics:
 
 preferences:
   show_search_suggestions:
-    type: string_list
+    type: boolean
```

Comment:
Just for your awareness: This will work, but land in a different column.
Old: `metrics.string_list.preferences_show_search_suggestions`
New: `metrics.boolean.preferences_show_search_suggestions`

That means comparing these across versions is a bit more involved, but not impossible.
(There should never be a client that has both set though)

(I'm still waiting for confirmation that what I said is actually true)

Reply:
Yes, please confirm that this is OK.  Since this workaround is because currently there are high number of startup metrics that have null values, I would argue that comparing against data we collected before is not very useful.

Reply:
Yeah, I agree it sounds like it's worthwhile to make the change.  You still may want to add in the description field "Prior to release X, this metric was of type `string_list`" so that *if* someone wanted to trace back through the (problematic) historical data they'd at least know how...

Reply:
This kind of change is not supported by our tooling, a similar change of a string to a datetime in the past has caused delays on our schema deploys: https://github.com/mozilla/mozilla-schema-generator/issues/118

Type changes is typically frowned upon at a low level because it [breaks schema evolution rules](https://cloud.google.com/bigquery/docs/managing-table-schemas) that allows for low operational overhead during ingestion time. It's possible to support changing types in the schema generator based on the behavior of moving the probe to a new namespace (which avoids a schema incompatible change). However, this will be confusing from a metadata perspective since we lose context on the type change.

I suggest expiring these probes and creating a new probe with the correct type, while clearly documenting the reason for deprecation so that tools like the [glean dictionary](https://dictionary.telemetry.mozilla.org/) can pick this up.

Reply:
@rocketsroger Can you ping me when you think you need a final review? Thanks!

Reply:
@gabrielluong Sure, this should be the final change but I'll ping you once I have the go ahead from the data science team.  Thanks,